lib/goth/config.ex

```diff
@@ -1,32 +0,0 @@
-defmodule Goth.Config do
```

Comment:
I appreciate that there's no real need for `Goth.Config` to be a `GenServer`, but I think we should keep this module around to allow consistent access to Goth's config. There are applications/dependencies that need access to the `project_id`, for instance, and right now, those retrieve that via `Goth.Config.get(:project_id)`, which feels more elegant to me than `Application.get_env(:goth, :config)[:project_id]`.

Even if `Goth.Config` does nothing more than call `Application.get_env/2` under the covers (which I think is fine), I'd prefer to keep it around for backwards compatibility.